include/nanobind/eigen/dense.h

```diff
@@ -241,6 +249,107 @@ struct type_caster<Eigen::Ref<T, Options, StrideType>> {
     operator Ref() { return Ref(caster.operator Map()); }
 };
 
+/// Caster for Eigen::SparseMatrix<T>
+template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = std::remove_reference_t<decltype(*std::declval<T>().outerIndexPtr())>;
```

Comment:
That definition of `StorageIndex` strikes me as surprisingly complicated. Can we not get the type more directly from a `using` alias defined within `T` (e.g. `T::StorageIndex`?)